janus-idp/backstage-showcase
# chore(auth): add default resolver for OIDC auth provider

## Description

Add back the old OIDC Auth Provider default resolver while maintaining the ability to declaratively configure the resolvers.

Also update docs for the oidc and guest auth provider sign in.

## Which issue(s) does this PR fix

- Fixes [RHIDP-2679](https://issues.redhat.com/browse/RHIDP-2679)

## PR acceptance criteria

Please make sure that the following steps are complete:


## How to test changes / Special notes to the reviewer

- Setup a keycloak instance, add a user with a username different from the locale part of it's email
  - ex: username: tester, email: test@mail.com
- Setup the keycloak-backend to ingest the user from the keycloak instance
- Setup OIDC auth provider without declarative resolver
  - Start the backstage instance and login with OIDC, you should observe a user not found due to mismatch of username and email
- Set the OIDC auth provider resolver to `preferredUsernameMatchingUserEntityName`
  - Start the backstage instance and login with OIDC, and you should be able to login as `user:default/tester`
